cmake/modules/FindQatDrv.cmake

```diff
@@ -0,0 +1,122 @@
+      	       "Please set an environment variable \"ICP_ROOT\" for the QAT driver package root directory, "
+	       "i.e. \"export ICP_ROOT=/the/directory/of/QAT\". "	
+	       "Or download and install QAT driver ${expected_version} manually at the link "
+	       "${expected_version_url}. "
+	       "Remember to set an environment variable for the package root directory, "
+               "i.e. \"export ICP_ROOT=/opt/APP/driver/QAT\"..")
+    else()
+      set(ENV{ICP_ROOT} ${expected_source_dir})
+    endif()
+  endif()
+endfunction()
+
+function(check_qatdrv_version expected_version expected_version_url)
+  string(REGEX REPLACE "^v1.7.L.([0-9]+.[0-9]+.[0-9]+).[0-9]+$"
+	 "\\1" EXPECTED_version "${expected_version}")
+  set(source_dir $ENV{ICP_ROOT})
+  set(versionfile "${source_dir}/versionfile")
+  if(NOT EXISTS ${versionfile})
+    message(FATAL_ERROR "${versionfile} not found. Please either \"rm -rf ${source_dir}\" "
+            "so you can download and install QAT driver ${expected_version} manually at the link "
+            "${expected_version_url}, or make sure ${source_dir} "
+            "contains a full copy of QAT driver ${expected_version}. "
+	    "Remember to set an environment variable for the package root directory, "
+	    "i.e. \"export ICP_ROOT=/opt/APP/driver/QAT\".")
+  endif()
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_TYPE=QAT[0-9]+.[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_TYPE=QAT([0-9]+.[0-9]+)$"
+         "\\1" QAT_VERSION1 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_OS=[A-Z]$")
+  string(REGEX REPLACE "^PACKAGE_OS=([A-Z])$"
+         "\\1" QAT_VERSION2 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_MAJOR_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_MAJOR_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION3 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_MINOR_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_MINOR_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION4 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_PATCH_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_PATCH_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION5 "${QAT_VERSION_LINE}")
+  unset(QAT_VERSION_LINE)
+  file(STRINGS "${versionfile}" QAT_VERSION_LINE
+       REGEX "^PACKAGE_VERSION_BUILD_NUMBER=[0-9]+$")
+  string(REGEX REPLACE "^PACKAGE_VERSION_BUILD_NUMBER=([0-9]+)$"
+         "\\1" QAT_VERSION6 "${QAT_VERSION_LINE}")
+  set(version_a "v${QAT_VERSION1}.${QAT_VERSION2}.${QAT_VERSION3}.${QAT_VERSION4}.${QAT_VERSION5}-${QAT_VERSION6}")
+  set(version "${QAT_VERSION3}.${QAT_VERSION4}.${QAT_VERSION5}")
+  if(version VERSION_LESS EXPECTED_version)
```

Comment:
why not pass `${version}` as the `VERSION_VAR` argument to `find_package_handle_standard_args()`?

Reply:
please reconsider this suggestion.